luaentity.lua

```diff
@@ -382,13 +382,36 @@ local move_entities_globalstep_part2 = function(dtime)
 	end
 end
 
-local handle_active_blocks_timer = 0.1
+-- dtime after which there is an update (or skip).
+local dtime_threshold = pipeworks.globalstep_interval
+-- Accumulated dtime since last update (or skip).
+local dtime_accum = 0
+-- Delayed dtime accumulated due to skipped updates.
+local dtime_delayed = 0
+local skip_update = false
 
 minetest.register_globalstep(function(dtime)
-	handle_active_blocks_timer = handle_active_blocks_timer + dtime
-	if dtime < 0.2 or handle_active_blocks_timer >= (dtime * 3) then
-		handle_active_blocks_timer = 0.1
-		move_entities_globalstep_part1(dtime)
-		move_entities_globalstep_part2(dtime)
+	if dtime >= 0.2 and dtime_delayed < 1 then
+		-- Reduce activity when the server is lagging.
+		skip_update = true
 	end
```

Comment:
You can't assume dtime is normally less than 0.2. See e.g. https://github.com/minetest-monitoring/mesecons_debug for how to track whether things are laggy or not, and whether that's your fault or not.

Reply:
I was just basing this part off the previous code. For all I know it isn't necessary.